### PR TITLE
fix: [CDS-58921]: oc version upgrade to 4.13.4

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Only for immutable delegate updating oc bindary to 4.13.4, latest 4.13 version
## Testing 
1. tested AMD binary:
https://qa.harness.io/ng/account/OgiB4-xETamKNVAz-wQRjw/cd/orgs/default/projects/DoNotDelete_Tarun_Bugathon/pipelines/rolllings/deployments/1AxD4I2lQSiU5T8a7nSONQ/pipeline?storeType=INLINE
Screenshot 2023-08-31 at 12 15 43 PM
2. tested ARM binary
https://qa.harness.io/ng/account/OgiB4-xETamKNVAz-wQRjw/cd/orgs/default/projects/DoNotDelete_Tarun_Bugathon/pipelines/rolllings/deployments/1AxD4I2lQSiU5T8a7nSONQ/pipeline?storeType=INLINE
Screenshot 2023-08-31 at 12 17 59 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51112)

### CODE REVIEW

**Comment by alebuh** [comment]
Do we need to upgrade only for immutable delegate or for legacy as well? I thought that we need to do it only for immutable?

**Review by GoelPratyush99 — CHANGES_REQUESTED** [review]
We still have these 2 files left to update the OC version 
Screenshot 2023-08-02 at 3 48 03 PM

**Comment by TarunUba1** [comment]
> Do we need to upgrade only for immutable delegate or for legacy as well? I thought that we need to do it only for immutable?

**Review by markoharness — CHANGES_REQUESTED** [review]
Shouldn't these classes also need to be updated?

io.harness.delegate.clienttools.OcVersion
io.harness.delegate.service.K8sGlobalConfigServiceImpl
dockerization/delegate-proxy/setup.sh
dockerization/delegate/scripts/client_tools.sh
on-prem/disconnected_on_prem_pov_installer/generate.sh
on-prem/dockerless_pov_installer_generator/generate_installer.sh
on-prem/harness_disconnected_on_prem_pov_final/install_harness.sh

and appropriate UT classes as well

**Review by bharatgoelharness — CHANGES_REQUESTED** [review]
did you tested this version for arm64 and amd64 both?

Please update the testing done as part of pr in description.

File: 960-api-services/src/main/java/io/harness/delegate/clienttools/OcVersion.java [hunk 1]
```diff
@@ -16,8 +16,8 @@
 @Getter
 @Slf4j
 public enum OcVersion implements ClientToolVersion {
-  V4_2("v4.2.16");
-
+  V4_2("v4.2.16"),
+  V4_13("v4.13.4");
```
Comment (alebuh) [inline 960-api-services/src/main/java/io/harness/delegate/clienttools/OcVersion.java:20]: Are we using this enum anywhere? If not why do we need to add it?

**Review by alebuh — CHANGES_REQUESTED** [review]

**Review by alebuh — DISMISSED** [review]

**Comment by TarunUba1** [comment]
trigger uts

**Comment by TarunUba1** [comment]
trigger commonchecks

**Comment by TarunUba1** [comment]
trigger uts

**Review by alebuh — APPROVED** [review]

**Review by GoelPratyush99 — APPROVED** [review]

**Comment by TarunUba1** [comment]
trigger review

**Pull request merged**